### DISCUSSION TITLE
Adding Page Sub Navigation Hooks

### DISCUSSION THREAD

**Discussion by dbpolito** [filamentphp/filament#14134] [Ideas]
I have a need to render something right before the sidebar sub-navigation, i proposed this PR: https://github.com/filamentphp/filament/pull/14101

For adding a sub-navigation before / after hooks.

**Comment by danharrin** [comment]
You could have sidebar on some pages and topbar on others, hence why I suggested making the hooks separate? If you have a mixture of both or a plugin that supports both?

**Reply by dbpolito** [reply]
So we would need to have 3 hooks? one per each type of sub nav? or maybe we can also pass the subNavigationPosition as scope to they can react to that?

**Reply by danharrin** [reply]
4, above and below each type?

**Reply by danharrin** [reply]
the scope is not required since its in the hook name, the scope should be the name of the page

**Reply by dbpolito** [reply]
i see... we have 3 types, plus de select (for mobile)...

Gotcha...

But you're not receiving any PR at the moment? Or if i open a PR using 4 hooks are you willing to merge now? Or wait for v4 to open for that branch?

**Reply by danharrin** [reply]
Ideally it would be for v4 so that I am spending less time reviewing new stuff for v3. If you make a perfect PR with all the documentation and everything I might be willing to merge it in v3, but if there are changes to be made you'll have to wait

**Reply by dbpolito** [reply]
I did the changes you asked, the PR needs to be reopened (which i can't) to update... you can see the new diff here:

https://github.com/filamentphp/filament/compare/3.x...dbpolito:filament:adding_page_sub_navigation_hooks?expand=1

Or should i open a new PR?

Just FYI, this isn't any blocker for me... i overwrote the package view in my project and made the changes i needed... this is just to give back to the community as these hooks are pretty useful and adds a lot of flexibility.

Up to you if you want to merge this on v3 or wait for v4... 🫶

**Reply by dbpolito** [reply]
If you prefer v4, is the 4.x open for new PRs? should i look there and make the changes there? 🤔

**Reply by danharrin** [reply]
Please rename `NAVIGATION_START` to `NAVIGATION_SIDEBAR_START` and the same with `END`. `NAVIGATION_TOP` can be `NAVIGATION_TABS`

**Reply by dbpolito** [reply]
my idea was to be consistent with the `SubNavigationPosition`, as they are called `Start`, `Top`, `End`... https://filamentphp.com/docs/3.x/panels/resources/getting-started#sub-navigation-position

No tabs / sidebar is mentioned in the docs as far as i can tell...

**Reply by danharrin** [reply]
Alright fair enough, merged #14101. Thanks

**Comment by dbpolito** [comment]
Solved in https://github.com/filamentphp/filament/pull/14101
